math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -848,6 +848,59 @@ RVec<RVec<typename RVec<T>::size_type>> Combinations(const RVec<T>& v, const typ
    }
 }
 
+/// Return the indices of the elements which are not zero
+template <typename T>
+RVec<typename RVec<T>::size_type> Nonzero(const RVec<T> &v)
+{
+   using size_type = typename RVec<T>::size_type;
+   RVec<size_type> r;
+   for(size_type i=0; i<v.size(); i++) {
+      if(v[i] != 0) {
+         r.emplace_back(i);
+      }
+   }
+   return r;
+}
+
+/// Return the intersection of elements of two vectors.
+/// Each element of v1 is looked up in v2 and added to the returned vector if
+/// found. Following, the order of v1 is reserved. Repetitions of elements in
+/// v2 are added multiple times to the returned vector if allow_repetitions is
+/// set to true;
+template <typename T>
+RVec<T> Intersect(const RVec<T>& v1, const RVec<T>& v2, bool allow_repetitions = false)
+{
+   // TODO: Remove comment after review.
+   // NOTE: This can be done much more efficient (for large vectors) if we sort
+   // v1 and v2 before.
+   RVec<T> r;
+   using size_type = typename RVec<T>::size_type;
+   for(size_type i=0; i<v1.size(); i++) {
```

Comment:
That's pretty greedy. Any good ideas to make this nicer? As long as we do not assume that `v1` and `v2` are sorted, I do not see any other solution than brute-forcing it.

Reply:
For the use-case that was raised on mattermost, I think only the ordering of `v1` is important to be conserved (ie. `v1` can be arbitrarily sorted, and the elements in the returned RVec `r` should follow that ordering), whereas the provided order of `v2` is not important.  As a result, we could first sort `v2` and then use a more efficient find than a linear scan (eg. bisect, etc), possibly using an STL algorithm.  If the user knows `v2` is sorted, we could provide an additional option to disable the initial sorting.  Alternatively, we can do what the STL does and just assume `v2` to be sorted and make sure the documentation is clear enough... Personally I think I'd prefer the first option.